declarations/AmdTemplatePlugin.d.ts

```diff
@@ -0,0 +1,25 @@
+export = AmdTemplatePlugin;
+/** @typedef {import("./Compiler")} Compiler */
+/**
+ * @typedef {Object} AmdTemplatePluginOptions
+ * @param {string=} name the library name
+ * @property {boolean=} requireAsWrapper
+ */
+declare class AmdTemplatePlugin {
+    /**
+     * @param {AmdTemplatePluginOptions} options the plugin options
+     */
+    constructor(options: any);
+    name: any;
+    requireAsWrapper: any;
+    /**
+     * @param {Compiler} compiler the compiler instance
+     * @returns {void}
+     */
+    apply(compiler: import("./Compiler")): void;
+}
+declare namespace AmdTemplatePlugin {
+    export { Compiler, AmdTemplatePluginOptions };
+}
+type Compiler = import("./Compiler");
+type AmdTemplatePluginOptions = any;
```

Comment:
It's _probably_ worth noting that this is `any` here because your doc -
```
/**
 * @typedef {Object} AmdTemplatePluginOptions
 * @param {string=} name the library name
 * @property {boolean=} requireAsWrapper
 */
```
doesn't make sense to us and is a kind of error. A `typedef` with both a `@param` and a `@property` doesn't make any sense.

Reply:
yeah, this seem to be a typo and `@param` should be `@property` here.